Add linter

danielgavrilov opened the pull request:
Fixes #8. Uses the [recommended rules](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.ts), I disabled all that were failing, although it's maybe a good idea to re-enable some?

eduhenke commented:
Hi @danielgavrilov, so sorry for the ultra-late response. I've discovered that I was not watching this project's changes :man_facepalming:.

We are more than happy to accept this PR, could you resolve the conflicts, so I can merge it?

Thanks for your contribution

danielgavrilov commented:
No worries @eduhenke! I just rebased, let me know if it all looks good, happy to make changes.

I left some warnings in there for unused variables – there's a rule to allow args/vars beginning with `_` to be unused, but others show a warning:

```
./src/integration.spec.ts
  17:29  warning  'cpId' is defined but never used. Allowed unused args must match /^_/u                @typescript-eslint/no-unused-vars
  33:39  warning  'req' is defined but never used. Allowed unused args must match /^_/u                 @typescript-eslint/no-unused-vars
  33:44  warning  'cpId' is defined but never used. Allowed unused args must match /^_/u                @typescript-eslint/no-unused-vars
  48:29  warning  'req' is defined but never used. Allowed unused args must match /^_/u                 @typescript-eslint/no-unused-vars
  51:29  warning  'req' is defined but never used. Allowed unused args must match /^_/u                 @typescript-eslint/no-unused-vars
  52:11  warning  'waitCpReq' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  55:48  warning  'chargePointId' is defined but never used. Allowed unused args must match /^_/u       @typescript-eslint/no-unused-vars

./src/ws/connection.ts
   5:50  warning  'ValidationError' is defined but never used. Allowed unused vars must match /^_/u          @typescript-eslint/no-unused-vars
   6:34  warning  'validateMessageResponse' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
   8:51  warning  'Maybe' is defined but never used. Allowed unused vars must match /^_/u                    @typescript-eslint/no-unused-vars
  96:20  warning  'action' is defined but never used. Allowed unused args must match /^_/u                   @typescript-eslint/no-unused-vars
  96:28  warning  'ocppVersion' is defined but never used. Allowed unused args must match /^_/u              @typescript-eslint/no-unused-vars
```

Let me know if you prefer to allow unused variables.

To lint run `yarn lint`. Also added linting in the GitHub Test action.
